numpy/lib/polynomial.py

```diff
@@ -678,7 +678,10 @@ def polyval(p, x):
         x = NX.asarray(x)
         y = NX.zeros_like(x)
```

Comment:
One more small request, if you fix these things, please also add a test, so that the next one changing this code can't accidentally break it.